demo/generate_notebooks.py

```diff
@@ -0,0 +1,75 @@
+import nbformat as nbf
+import os 
+import json
+import random
+
+GRADIO_DEMO_DIR = os.getcwd()
+DEMOS_TO_SKIP = {"all_demos", "reset_components", "custom_path", "kitchen_sink_random"}
```

Comment:
Why are we skipping "reset_components", "custom_path", "kitchen_sink_random"

Reply:
`reset_components` and `kitchen_sink_random` import from other demos so can't be used on colab alone. We also skip them when uploading to spaces because they don't work there either. They're only used for the PR demos. (along with `all_demos`)

`custom_path` is a demo of  launching gradio using uvicorn from the terminal, so not really useful on colab either (also skipped on spaces)